hbase-server/src/main/java/org/apache/hadoop/hbase/wal/AbstractWALRoller.java

```diff
@@ -148,10 +148,9 @@ private void abort(String reason, Throwable cause) {
   @Override
   public void run() {
     while (running) {
-      boolean periodic = false;
       long now = System.currentTimeMillis();
       checkLowReplication(now);
-      periodic = (now - this.lastRollTime) > this.rollPeriod;
+      boolean periodic = (now - this.lastRollTime) > this.rollPeriod;
```

Comment:
Not on this patch directly.
We have the periodic WAL roll.. Every WAL should get rolled as per this period.  When one WAL gets rolled, we will change the lastRollTime and so it can happen that some WALs will ever get rolled!
This issue will be visible now as we will selectively roll WAL files after this patch.  Actually we need to track the lastRollTime per WAL instance.

Reply:
It's like this, I ignored it